app/services/digital_links_creator.rb

```diff
@@ -0,0 +1,26 @@
+class DigitalLinksCreator
+  attr_reader :line_item
+
+  delegate :quantity, :digital_links, :variant, :master, to: :line_item
+
+  def initialize(line_item)
+    @line_item = line_item
+  end
+
+  def create_digital_links
+    digital_links.delete_all
+
+    #include master variant digitals
+    master = variant.product.master
+    create_digital_links_for_variant(master) if master.digital?
+    create_digital_links_for_variant(variant) unless variant.is_master
+  end
+
+  private
+    def create_digital_links_for_variant(variant)
+      variant.digitals.each do |digital|
+        digital.create_drm_attachment(line_item) if digital.drm?
+        quantity.times { digital_links.create!(digital: digital) }
```

Comment:
Was in original gem, no idea why, but I didn't want to break backwards compatibility